src/flood_wave_detector.py

```diff
@@ -82,6 +83,8 @@ def find_vertices(self) -> None:
         The end result is saved to 'PROJECT_PATH/generated/find_vertices' folder.
         :return:
         """
+        g = open(os.path.join(PROJECT_PATH, "data", "level_groups_fontos.json"))
```

Comment:
I would put this file load to `find_local_maxima`, so `find_local_maxima` will not have a `level_group` argument.